packages/mdc-checkbox/_mixins.scss

```diff
@@ -0,0 +1,400 @@
+@import "@material/rtl/mixins";
+@import "./functions";
+@import "./keyframes";
+@import "./variables";
+
+//
+// Public
+//
+
+// TODO(acdvorak): Figure out why a pre-checked checkbox janks when `indeterminate` is toggled for the first time.
+// TODO(acdvorak): Accessibility
+// TODO(acdvorak): Do we need [aria-disabled="true"] selectors?
+
+// NOTE: Child selector MUST begin with `.mdc-checkbox__background`
+@mixin mdc-checkbox-background-unfilled-enabled {
+  .mdc-checkbox__native-control:not(:disabled) ~ {
+    @content;
+  }
+}
+
+// NOTE: Child selector MUST begin with `.mdc-checkbox__background`
+@mixin mdc-checkbox-background-unfilled-disabled {
+  .mdc-checkbox__native-control:disabled ~ {
+    @content;
+  }
+}
+
+// NOTE: Child selector MUST begin with `.mdc-checkbox__background`
+@mixin mdc-checkbox-background-filled-enabled {
+  .mdc-checkbox__native-control:not(:disabled):checked ~,
+  .mdc-checkbox__native-control:not(:disabled):indeterminate ~ {
+    @content;
+  }
+}
+
+// NOTE: Child selector MUST begin with `.mdc-checkbox__background`
+@mixin mdc-checkbox-background-filled-disabled {
+  .mdc-checkbox__native-control:disabled:checked ~,
+  .mdc-checkbox__native-control:disabled:indeterminate ~ {
+    @content;
+  }
+}
+
+@mixin mdc-checkbox-stroke-color($color) {
+  .mdc-checkbox__background {
+    @include mdc-theme-prop(border-color, $color);
+  }
+}
+
+@mixin mdc-checkbox-container-color($color) {
+  .mdc-checkbox__background {
+    @include mdc-theme-prop(background-color, $color);
+  }
+}
+
+@mixin mdc-checkbox-focus-color($color) {
+  // The ::before element is used as a focus indicator for the checkbox
+  .mdc-checkbox__background::before {
+    @include mdc-theme-prop(background-color, $color, $edgeOptOut: true);
+  }
+}
+
+@mixin mdc-checkbox-ink-color($color) {
+  // TODO(acdvorak): Rename to .mdc-checkbox__checkmark-path
+  .mdc-checkbox__checkmark__path {
+    // !important is needed here because a stroke must be set as an attribute on the SVG in order
+    // for line animation to work properly.
+    @include mdc-theme-prop(stroke, $color, $important: true);
+  }
+
+  .mdc-checkbox__mixedmark {
+    @include mdc-theme-prop(background-color, $color);
+  }
+}
+
+@mixin mdc-checkbox-ripple-color($color, $opacity: .14) {
+  // TODO(acdvorak): Refactor ripple SCSS to so that we can override color only.
+
+  $ripple-config: (base-color: $color);
+
+  // TODO(acdvorak): Refactor ripple mixin to do this @if check automatically, or use mdc-theme-prop()
+  @if (map-has-key($mdc-theme-property-values, $color)) {
+    $ripple-config: (theme-style: $color);
+  }
+
+  $ripple-config: map-merge((opacity: $opacity), $ripple-config);
+
+  @include mdc-ripple-bg(map-merge((pseudo: "::before"), $ripple-config));
+  @include mdc-ripple-fg(map-merge((pseudo: "::after"), $ripple-config));
+
+  &::before,
+  &::after {
+    @include mdc-checkbox-ripple-size_;
+  }
+}
+
+@mixin mdc-checkbox-state-transition-color($color) {
```

Comment:
Done.

For consistency, I still allow callers to pass a single argument for both colors, because I doubt most people will want separate container and stroke colors for a marked checkbox (though I need to confirm that with design). You can disable that behavior by explicitly passing `false` for `$stroke-color`.

Reply:
I think I would make both container-color and stroke-color required...its a simpler API that way. Otherwise `mdc-checkbox-check-animation` might be a little too magical.

Reply:
Done